pkg/util/admission/work_queue.go

```diff
@@ -1523,6 +1514,33 @@ func addName(name string, meta metric.Metadata) metric.Metadata {
 // shared across WorkQueues, so Gauges should only be updated using deltas
 // instead of by setting values.
 type WorkQueueMetrics struct {
+	name       string
+	total      workQueueMetricsSingle
+	byPriority sync.Map
+	registry   *metric.Registry
+}
+
+// getOrCreate will return the metric if it exists or create it and then return
+// it if it didn't previously exist.
+func (m *WorkQueueMetrics) getOrCreate(priority admissionpb.WorkPriority) workQueueMetricsSingle {
+	// Try loading from the map first. This will not fin
+	val, ok := m.byPriority.Load(priority)
+	if !ok {
+		// This will only happen the first time it is requested. Doing this lazily
+		// prevents unnecessary creation of unused priorities. Note that it is
+		// necessary to call LoadOrStore here as this could be called concurrently.
+		// It is not called the first Load so that we don't have to unnecessarily
+		// create the metrics.
+		statPrefix := fmt.Sprintf("%v.%v", m.name, priority)
+		val, ok = m.byPriority.LoadOrStore(priority, makeWorkQueueMetricsSingle(statPrefix))
+		if !ok {
+			m.registry.AddMetricStruct(val)
```

Comment:
With this lazy registration -- does these metrics show up in CRDB's internal timeseries database? I'm asking whether it works with everything under pkg/server/status/recorder.go since a lot of that code was written with upfront registration in mind.